test/fuzz/CMakeLists.txt

```
@@ -0,0 +1,17 @@
+tests(GROUPS fuzz
+      SOURCES
+      simplify.cpp
+)
+
+
+target_link_libraries(fuzz_simplify PRIVATE Halide::Halide)
+
+# Allow OSS-fuzz to manage flags directly
+if (DEFINED ENV{LIB_FUZZING_ENGINE})
```

Please do not use environment variables this way in CMake. Unlike with autotools, the user does not control when the build system is regenerated; the build system itself does according to CMake's analysis of the build file dependencies, which includes CMake's own implementation. This will cause vexing behavior when a developer upgrades CMake. Imagine:

> I upgraded CMake and now the fuzz tests aren't building! Why? Oops, I didn't set `LIB_FUZZING_ENGINE` in Bash again.

Thus, cache variables should always dominate. To initialize a cache variable from the environment, write:

```suggestion
set(LIB_FUZZING_ENGINE "$ENV{LIB_FUZZING_ENGINE}"
    CACHE STRING "Compiler flags necessary to use libFuzzer instrumentation")

if (LIB_FUZZING_ENGINE)
```

---

Cool thanks for the suggestion. I'll add that in. I'm definitely a cmake noob, so I appreciate the guidance.

---

> I'm definitely a cmake noob,

For your own sanity, stay that way 😆

---

> > I'm definitely a cmake noob,
> 
> For your own sanity, stay that way 😆

lol yeah, I find cmake in general just... baffling